packages/compiler-cli/ngcc/src/host/esm2015_host.ts

```diff
@@ -353,6 +353,25 @@ export class Esm2015ReflectionHost extends TypeScriptReflectionHost implements N
       }
     }
 
+    // We detect if the resolved declaration refers to the JavaScript global `Object` by checking
+    // if the declaration node is named `Object`, resolves to a type `ObjectConstructor`, and has
+    // been defined in a default library source file.
+    if (superDeclaration && superDeclaration.node !== null &&
```

Comment:
In the compiler we tend to use explicit `null`/`undefined` checks, even if's more verbose.

```suggestion
    if (superDeclaration !== null && superDeclaration.node !== null &&
```

And yes, ngcc doesn't follow this convention very closely 😅